### DISCUSSION TITLE
Using python.exe as target instead of pythonw.exe

### DISCUSSION THREAD

**Discussion by Nanguage** [beeware/briefcase#825] [Q&A]
Hi, I have a question about the msi installer.

When building a windows msi installer. How can I use 'python.exe' instead of 'pythonw.exe' as the target of the shortcut?

**Comment by freakboy3742** [comment]
You can't - at least, there's no flag or option to enable that behavior. 

On top of that, the next version of Briefcase will change the Windows template such that pythonw isn't used either, replacing it with a stub app.

Can I ask *why* you want to do this?

**Reply by Nanguage** [reply]
Okay, I look forward to the new version. I want my app show the terminal window to display some log. But the 'pythonw' suppresses it.

**Reply by freakboy3742** [reply]
Ah - if that's the case, the Briefcase release that we've just published (0.3.9) may be of interest to you, as we've restructured the way Windows apps are started so that console output is visible (and app crashes are caught with a dialog).

**Reply by mhsmith** [reply]
Specifically, the new Briefcase version now shows the console output of Windows apps when using `briefcase run`, but I don't think it will show a terminal window in apps which have been installed using the MSI. You can still launch the installed executable *within* a terminal window if you want to see the logs.

**Reply by freakboy3742** [reply]
Yes - to clarify - the log will be visible in the console if you start the app from the console; however, if you start the app by clicking on an icon in the start menu (or filesystem), any console output will be written to a file in your user directory, with logs being rotated on every execution, and the last 10 log files being retained.

**Comment by Nanguage** [comment]
I have found a temporary solution. After run `birefcase create`. There are `xxx.wxs` file generated in `windows/msi/xxx`.  Open it with text editor, then `Ctrl F` search the keyword `pythonw.exe`, replace it with `python.exe`.

![image](https://user-images.githubusercontent.com/14010939/184885063-d03de4b6-c745-4248-bb20-f9c7def9e047.png)

Then compile the app with `birefcase package`. The target will be replaced with `python.exe`.
